src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs

```diff
@@ -127,15 +127,51 @@ public async Task<OpenApiDocument> GenerateAsync(ApiDescriptionGroupCollection a
 
         /// <summary>Gets the default serializer settings representing System.Text.Json.</summary>
         /// <returns>The settings.</returns>
-        public static JsonSerializerSettings GetSystemTextJsonSettings()
+        public static JsonSerializerSettings GetSystemTextJsonSettings(IServiceProvider serviceProvider)
         {
             // If the ASP.NET Core website does not use Newtonsoft.JSON we need to provide a 
             // contract resolver which reflects best the System.Text.Json behavior.
             // See https://github.com/RicoSuter/NSwag/issues/2243
-            return new JsonSerializerSettings
+            JsonSerializerSettings settings = new JsonSerializerSettings
             {
                 ContractResolver = new CamelCasePropertyNamesContractResolver()
             };
+
+#if NETSTANDARD2_0
+            if (serviceProvider == null)
+            {
+                return settings;
+            }
+                
+            dynamic options = null;
+            try
+            {
+                var optionsAssembly = Assembly.Load(new AssemblyName("Microsoft.AspNetCore.Mvc.Core"));
+                var optionsType = typeof(IOptions<>).MakeGenericType(optionsAssembly.GetType("Microsoft.AspNetCore.Mvc.JsonOptions", true));
+                options = serviceProvider?.GetService(optionsType) as dynamic;
+            }
+            catch
+            {
+            }
+
+            if (!(options?.Value?.JsonSerializerOptions is System.Text.Json.JsonSerializerOptions jsonOptions))
+            {
+                return settings;
+            }
+
+            // StringEnumConverter
+            if (jsonOptions.Converters.Any(c => c is System.Text.Json.Serialization.JsonStringEnumConverter))
+            {
+                settings.Converters.Add(new Newtonsoft.Json.Converters.StringEnumConverter());
```

Comment:
Maybe we should also support this new attribute here: 
https://github.com/RicoSuter/NJsonSchema/blob/31d67d7276b88074f472377f94cd1fa8647b2953/src/NJsonSchema/Generation/DefaultReflectionService.cs#L386

Reply:
Looks like the System.Text.Json attribute is similar enough that adding an IsAssignableToTypeName for the new converter name should do the trick. I could test it and make a pull request

Reply:
Exactly should be just an or with another check, maybe with namespace. PR would be great.

Reply:
Done.